src/bindings/ScriptDev2/scripts/outland/black_temple/boss_reliquary_of_souls.cpp View
@@ -157,7 +157,7 @@ struct MANGOS_DLL_DECL boss_reliquary_of_soulsAI : public Scripted_NoMovementAI
  	void MoveInLineOfSight(Unit* pWho) override
  	{
  		if (m_uiPhase == PHASE_0_NOT_BEGUN && pWho->GetTypeId() == TYPEID_PLAYER && !((Player*)pWho)->isGameMaster() &&
- 			m_creature->IsWithinDistInMap(pWho, m_creature->GetAttackDistance(pWho)) && m_creature->IsWithinLOSInMap(pWho))
+ 			m_creature->IsWithinDistInMap(pWho, 65.0f) && m_creature->IsWithinLOSInMap(pWho))
  		{
  			// Start phase 1
  			m_uiPhase = PHASE_1_SUFFERING;
 @@ -189,6 +189,9 @@ struct MANGOS_DLL_DECL boss_reliquary_of_soulsAI : public Scripted_NoMovementAI
  		case NPC_ESSENCE_ANGER:
  			DoScriptText(ANGER_SAY_FREED, pSummoned);
  			break;
+ 		case NPC_ENSLAVED_SOUL:
+ 			m_uiSoulSummonedCount++;
+ 			break;
  		}
  
  		// All summons are set in combat
 @@ -234,7 +237,6 @@ struct MANGOS_DLL_DECL boss_reliquary_of_soulsAI : public Scripted_NoMovementAI
  	void DoNotifySouldDead()
  	{
  		++m_uiSoulDeathCount;
  		// Prepare to summon the essence
  		if (m_uiSoulDeathCount == MAX_ENSLAVED_SOULS)
  		{
@@ -306,7 +308,6 @@ struct MANGOS_DLL_DECL boss_reliquary_of_soulsAI : public Scripted_NoMovementAI
  				{
  					if (DoCastSpellIfCan(m_creature, SPELL_SUMMON_ENSLAVED_SOUL) == CAST_OK)
  					{
- 						++m_uiSoulSummonedCount;
  						m_uiSummonSoulTimer = 500;
  					}
  				}




src/bindings/ScriptDev2/scripts/outland/black_temple/boss_teron_gorefiend.cpp View
@@ -140,6 +140,7 @@ struct MANGOS_DLL_DECL boss_teron_gorefiendAI : public ScriptedAI
      {
      case NPC_SHADOWY_CONSTRUCT:
        pSummoned->CastSpell(pSummoned, SPELL_SHADOWY_CONSTRUCT, true);
+       pSummoned->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
        pSummoned->SetInCombatWithZone();
        break;
      case NPC_DOOM_BLOSSOM: